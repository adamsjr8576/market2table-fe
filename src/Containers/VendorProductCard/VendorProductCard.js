import React, { useState } from 'react';
import PropTypes from 'prop-types';
import './VendorProductCard.scss';

export const VendorProductCard = ({ products, setProducts, name, price, description, id }) => {

  const handleDelete = () => {
    const filteredProducts = products.filter(product => product.id !== id);
    setProducts(filteredProducts);
  }

  return (
    <article className='product-card-article'>
      <section className='product-card-section'>
        <p className='product-card-p'>{name}</p>
        <p className='product-card-p'>${price}</p>
      </section>
      <p className='product-card-description'>{description}</p>
      <button type='button' className='delete-product-button' onClick={handleDelete}>X</button>
    </article>
  )
}

export default VendorProductCard;

VendorProductCard.propTypes = {
  history: PropTypes.object,
  products: PropTypes.array,
  setProducts:PropTypes.func,
  name: PropTypes.string,
  price: PropTypes.number,
  description: PropTypes.string,
  id: PropTypes.string
}
