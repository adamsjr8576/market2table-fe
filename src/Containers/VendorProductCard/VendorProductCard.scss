@import './src/variables';

.product-card-article {
  width: 95%;
  display: flex;
  align-items: center;
  justify-content: space-around;
  border: 2px solid black;
  border-radius: 20px;
  height: 50px;
  margin: 8px 0;
  border: 2px solid $dark_green_border_color;
  background-color: $list_hover_color;
}

.product-card-section {
  width: 35%;
  text-align: center;
  height: 90%;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  border-right: 2px solid $dark_green_border_color;
}

.product-card-p {
  margin: 2px;
}

.product-card-description {
  width: 55%;
}

.delete-product-button {
  width: 20px;
  height: 20px;
  border-radius: 50%;
  text-align: center;
  padding: 0;
  margin-right: 3px;
  cursor: pointer;
  border: 1px solid black;
  outline: 0;
  background-color: white;
}

.delete-product-button:hover {
  color: red;
  border: 1px solid red;
}

.delete-product-button:active {
  transform: scale(.95);
}
