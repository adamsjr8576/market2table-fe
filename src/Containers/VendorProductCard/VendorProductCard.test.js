import React from 'react';
import { shallow } from 'enzyme';
import { VendorProductCard } from './VendorProductCard';

describe('VendorProductCard', () => {

  let wrapper;
  let mockProducts = [{name: 'plain ol dough', price: '5', description: 'some good white bread', id: '2342'}];
  let mocksetProducts = jest.fn();
  let mockName = 'plain ol dough';
  let mockPrice = '6';
  let mockDescription = 'yummy food';
  let mockId = '23423';

  it('should match the snapshot', () => {
    wrapper = shallow(<VendorProductCard
      products={mockProducts}
      setProducts={mocksetProducts}
      name={mockName}
      price={mockPrice}
      description={mockDescription}
      id={mockId}
    />);

    expect(wrapper).toMatchSnapshot();
  });
});
