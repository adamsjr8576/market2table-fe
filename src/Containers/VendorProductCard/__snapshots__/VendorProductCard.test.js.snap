// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VendorProductCard should match the snapshot 1`] = `
<article
  className="product-card-article"
>
  <section
    className="product-card-section"
  >
    <p
      className="product-card-p"
    >
      plain ol dough
    </p>
    <p
      className="product-card-p"
    >
      $
      6
    </p>
  </section>
  <p
    className="product-card-description"
  >
    yummy food
  </p>
  <button
    className="delete-product-button"
    onClick={[Function]}
    type="button"
  >
    X
  </button>
</article>
`;
